bookserver/routers/assessment.py

```diff
@@ -76,3 +79,68 @@ async def get_assessment_results(
     #       res.update(res_update)
     rslogger.debug(f"Returning {row}")
     return make_json_response(detail=row)
+
+
+# Define a simple model for the gethist request.
+# If you just try to specify the two fields as parameters it expects
+# them to be in a query string.
+class HistoryRequest(BaseModel):
+    # ``acid`` : id of the active code block also called div_id
+    acid: str
+    # ``sid``: optional identifier for the owner of the code (username)
+    sid: Optional[str] = None
+
+
+@router.post("/gethist")
+async def get_history(request: Request, request_data: HistoryRequest):
+    """
+    return the history of saved code by this user for a particular
+    active code id (acid) -- known as div_id elsewhere
+    See :ref:`addHistoryScrubber`
+
+    :Parameters:
+        - See HistoryRequest
+
+    :Return:
+        - json object with a detail key that references a dictionary
+
+        ::
+
+            { "acid": div_id,
+              "sid" : id of student requested,
+              "history": [code, code, code],
+              "timestamps": [ts, ts, ts]
+            }
+    """
+    acid = request_data.acid
+    sid = request_data.sid
+    # if request_data.sid then we know this is being called from the grading interface
+    # so verify that the actual user is an instructor.
+    if sid:
+        if request.state.user and request.state.user.username != sid:
+            if await is_instructor(request):
+                course_id = request.state.user.course_id
+            else:
+                raise HTTPException(401)
+        else:
+            raise HTTPException(401)
+    # In this case, the request is simply from a student, so we will use
+    # their logged in username
+    elif request.state.user:
+        sid = request.state.user.username
+        course_id = request.state.user.course_id
+    else:
+        raise HTTPException(401)
+
+    res: Dict[str, Any] = {}
+    if sid:
```

Comment:
Will this ever be false?

Reply:
I guess not, now that we are using the HTTPException inthe else clause.